bundles/AdminBundle/Resources/public/js/pimcore/object/helpers/customLayoutEditor.js

```diff
@@ -792,45 +810,118 @@ pimcore.object.helpers.customLayoutEditor = Class.create({
-                params: {
-                    name: value,
-                    classId: this.klass.id
+        var identifierField = new Ext.form.field.Text({
+            fieldLabel: t('unique_identifier'),
+            labelWidth: 200,
+            maxLength: 20,
+            value: suggestedIdentifier
+        });
+
+        this.win = new Ext.Window({
+            title: t('add_layout'),
+            width: 400,
+            height: 250,
+            draggable: false,
+            border: false,
+            modal: true,
+            bodyStyle: "padding: 10px;",
+            resizable: true,
+            buttonAlign: 'center',
+            items: [
+                nameField,
+                identifierField, {
+                    xtype: 'panel',
+                    html: t('unique_identifier_warning')
+                }
+            ],
+            buttons: [
+                {
+                    xtype: 'button',
+                    text: t('cancel'),
+                    iconCls: 'pimcore_icon_cancel',
+                    handler: function () {
+                        this.win.close();
+
+                    }.bind(this)
                 },
-                success: function (response) {
-
-                    var data = Ext.decode(response.responseText);
-                    if(data && data.success) {
-                        this.setCurrentNode("root");
-                        this.editPanel.removeAll();
-                        this.classDefinitionPanel.enable();
-                        this.enableButtons();
-                        this.layoutComboStore.reload();
-                        this.currentLayoutId = data.id;
-                        this.layoutChangeCombo.setValue(data.id);
-                        this.initLayoutFields(true, response);
-                    } else {
-                        Ext.Msg.alert(t('error'), t('saving_failed'));
-                    }
-                }.bind(this)
-            });
+                {
+                    xtype: 'button',
+                    text: t('OK'),
+                    iconCls: 'pimcore_icon_save',
+                    handler: function ( nameField, identifierField, layouts, button) {
+                        if (this.addLayoutComplete(nameField.getValue(), identifierField.getValue(), layouts)) {
+                            this.win.close();
+                        }
+                    }.bind(this, nameField, identifierField, layouts)
+                }
+            ]
+        })
+        this.win.show();
+        nameField.focus();
+    },
+
+    addLayoutComplete: function (layoutName, layoutIdentifier, layouts) {
+
+        var layoutNameRegresult = layoutName.match(/[a-zA-Z][a-zA-Z0-9]+/);
+
+        if (layoutName.length <= 2 || layoutNameRegresult != layoutName) {
+            Ext.Msg.alert(' ', t('invalid_name'));
+            return false;
+        }
+
+        if (pimcore.helpers.findCaseInsensitiveValue(layouts["existingNames"], layoutName)) {
```

Comment:
Please use `in_arrayi()` instead which seems to do exactly the same as `pimcore.helpers.findCaseInsensitiveValue()`. 
Don't forget to remove the declaration of `pimcore.helpers.findCaseInsensitiveValue` as well :)